lib/vector-core/src/config/mod.rs

```diff
@@ -154,6 +154,8 @@ pub struct AcknowledgementsConfig {
 }
 
 impl AcknowledgementsConfig {
+    pub const DEFAULT: Self = Self { enabled: None };
```

Comment:
From what I can tell, this is just used for testing. Maybe we could call it `TEST` instead to make it clearer it shouldn't be used by a component?

Reply:
Hmm, I had actually defined the `Default` implementation to return `::DEFAULT` at one point. I had actually set up another constant like this to be used in `const` contexts where `::default()` can't be used.

Reply:
Ah, I see 🤔 My main concern is that we accidentally use the default rather than pulling it from the user provided configuration, but this seems like a fairly unlikely error.